stream_alert/rule_processor/handler.py

```diff
@@ -133,6 +160,69 @@ def get_alerts(self):
         """
         return self._alerts
 
+    def _send_to_firehose(self):
+        """Put all classified records into its respective Firehose Delivery Stream
+
+        Args:
+            categorized_payloads (defaultdict): Categorized set of payloads by classified type
+        """
+        def _chunk(record_list, chunk_size):
+            """Helper function to chunk payloads"""
+            for item in range(0, len(record_list), chunk_size):
+                yield record_list[item:item + chunk_size]
+
+        def _check_record_batch(batch):
+            """Helper function to verify record size"""
+            for index, record in enumerate(batch):
+                if len(str(record).encode('utf-8')) > MAX_RECORD_SIZE:
+                    # Show the first 1k bytes in order to not overload
+                    # CloudWatch logs
+                    LOGGER.error('The following record is too large'
+                                 'be sent to Firehose: %s', str(record)[:1000])
+                    MetricLogger.log_metric(FUNCTION_NAME,
+                                            MetricLogger.FIREHOSE_FAILED_RECORDS,
+                                            1)
+                    batch.pop(index)
+
+        delivery_stream_name_pattern = 'streamalert_data_{}'
+
+        # Iterate through each payload type
+        for log_type, records in self.categorized_payloads.items():
+            # This same method is used when naming the Delivery Streams
+            formatted_log_type = log_type.replace(':', '_')
+
+            for record_batch in _chunk(records, MAX_BATCH_SIZE):
+                stream_name = delivery_stream_name_pattern.format(formatted_log_type)
+                _check_record_batch(record_batch)
+
+                resp = self.firehose_client.put_record_batch(
+                    DeliveryStreamName=stream_name,
+                    Records=[{'Data': json.dumps(record, separators=(",", ":")) + '\n'}
+                             for record
+                             in record_batch])
+
+                # Error handle if failures occured
+                # TODO(jack) implement backoff here once the rule processor is split
+                if resp.get('FailedPutCount') > 0:
+                    failed_records = [failed
+                                      for failed
+                                      in resp['RequestResponses']
+                                      if failed.get('ErrorCode')]
+                    MetricLogger.log_metric(FUNCTION_NAME,
+                                            MetricLogger.FIREHOSE_FAILED_RECORDS,
+                                            resp['FailedPutCount'])
+                    LOGGER.error('The following records failed to Put to the'
+                                 'Delivery stream %s: %s',
+                                 stream_name,
+                                 json.dumps(failed_records, indent=2))
```

Comment:
Can we do a check here to make sure we aren't trying to dump 5k failed records to the logger before doing it? Maybe add an error log with the total failed record count and then json dump a subset (first 5/last 5) or something?